connectors/ssh/src/main/java/org/crsh/ssh/term/SSHLifeCycle.java

```diff
@@ -157,9 +157,9 @@ public void init() {
       server.setCompressionFactories(SSHFactories.setUpCompressionFactories());
 
       //
-      ArrayList<NamedFactory<Command>> namedFactoryList = new ArrayList<NamedFactory<Command>>(0);
+      ArrayList<SubsystemFactory> namedFactoryList = new ArrayList<>(0);
       for (SubsystemFactoryPlugin plugin : context.getPlugins(SubsystemFactoryPlugin.class)) {
-        namedFactoryList.add(plugin.getFactory());
+        namedFactoryList.add((SubsystemFactory)plugin.getFactory());
```

Comment:
Is anyone implementing SubsystemFactoryPlugin anywhere? Presumably not if our code is compiling ok.
But I think you need to at least change the return type of the getFactory() method in SubsystemFactoryPlugin to return a SubsystemFactory. Otherwise this cast will fail at runtime.